### DISCUSSION TITLE
dplyr::tbl when used on CSV could pass ... into duckdb's read_csv

### DISCUSSION THREAD

**Discussion by jangorecki** [duckdb/duckdb#7865] [Q&A] [answered]
Reading documentation on https://duckdb.org/docs/api/r we see that

> When using dbplyr, CSV and Parquet files can be read using the dplyr::tbl function.

and there is a following code exam

```r
write.csv(mtcars, "mtcars.csv")

tbl(con, "mtcars.csv") |>
  group_by(cyl) |>
  summarise(across(disp:wt, .fns = mean)) |>
  collect()
```

I would like to be able to pass extra arguments via `...` in `tbl()` to duckdb csv reader. Looking at the source in https://github.com/duckdb/duckdb/blob/e9b683ce153659dc6fc7983417e562f90e12ed42/tools/rpkg/R/backend-dbplyr__duckdb_connection.R#L392C5-L397
it is not clear if it is supported. If it is, then documentation would be very useful.

So for now it seems I have to rely on automatic detection of parameters by duckdb csv reader. List of parameters is so long, https://duckdb.org/docs/data/csv/overview, that there should be possibility to specify them when using `tbl()`.

**Comment by ghost** [comment, marked as answer]
Since `tbl` simply embeds the string in SQL, it seems to work when written as follows:

``` r
con <- DBI::dbConnect(duckdb::duckdb(), ":memory:")

mtcars |>
  tibble::as_tibble() |>
  dplyr::select(cyl) |>
  readr::write_csv("mtcars.csv")

dplyr::tbl(con, "read_csv('mtcars.csv', delim=',', header=True, columns={'cyl': 'int'})") |>
  dplyr::collect()
#> # A tibble: 32 × 1
#>      cyl
#>    <int>
#>  1     6
#>  2     6
#>  3     4
#>  4     6
#>  5     8
#>  6     6
#>  7     8
#>  8     4
#>  9     4
#> 10     6
#> # ℹ 22 more rows
```

<sup>Created on 2023-06-08 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Perhaps it seems worth documenting.

**Reply by jangorecki** [reply]
Definitely worth documenting! Otherwise is considered not officially supported. Thank you

**Comment by rsund** [comment]
You can actually use any (suitable) DuckDB function or sql in dplyr style with `tbl`.

And there are quite many features without full documentation in DuckDB. Therefore it may be a good idea to check also unit tests for possibilities, i.e. for R specific things the testthat directory in DuckDB R client.

I also believe that the core team is happy to accept pull requests to the [documentation repository](https://github.com/duckdb/duckdb-web) if you want to help.

**Reply by jangorecki** [reply]
So good we pointed out one such function so it can be improved now. I would normally file GH issue for that but duckdb has some more restricting policies about GH issues, therefore writing it here, hopefully will not be lost

**Reply by eitsupi** [reply]
I have sent a PR duckdb/duckdb-web#858
